codegen/arrai/svc_types.arrai

```diff
@@ -50,21 +50,35 @@ let orderedTypes = \types
                     cond {optParams: $`func (t *${typename}) UnmarshalJSON(data []byte) error {
                         inner := struct {
                             ${entity('attrDefs') where sysl.isPtr(.@value) orderby .@ >> \(@: key, @value: attrDef)
-                                $'${go.name(key)} ${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
+                                let jsonTag = 
+                                    cond attrDef {
+                                        {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
+                                        _: key,
+                                    };
+                                $'${go.name(key)} ${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
                             ::\i}
                             ${entity('attrDefs') where !sysl.isPtr(.@value) orderby .@ >> \(@: key, @value: attrDef)
-                                $'${go.name(key)} *${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
+                                let jsonTag = 
+                                    cond attrDef {
+                                        {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
+                                        _: key,
+                                    };
+                                $'${go.name(key)} *${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
```

Comment:
This condition is repeated four times. Can you pull this out into a function:
```
cond attrDef {
    {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
    _: key,
}
```

Reply:
Extracted the duplicates into a new code block.